apps/juxtaposition-ui/src/metrics.ts

```diff
@@ -0,0 +1,93 @@
+import { Gauge } from 'prom-client';
+import expressMetrics from 'express-prom-bundle';
+import express from 'express';
+import { logger } from '@/logger';
+import { config } from '@/config';
+import { SETTINGS } from '@/models/settings';
+import type { Express } from 'express';
+
+export const onlineNowGauge = new Gauge({
+	name: 'juxtaposition_online_users_now',
+	help: 'Users active in the last 10 minutes',
+	async collect(): Promise<void> {
+		const onlineRangeMs = 10 * 60 * 1000; // 10 minutes
+		const cutoff = new Date(Date.now() - onlineRangeMs);
+		const count = await SETTINGS.countDocuments({ last_active: { $gt: cutoff } });
+		this.set(count);
+	}
+});
+
+export const activeMonthlyGauge = new Gauge({
+	name: 'juxtaposition_active_users_monthly',
+	help: 'Users active in the last 30 days',
+	async collect(): Promise<void> {
+		const monthlyRangeMs = 30 * 24 * 60 * 60 * 1000;
+		const cutoff = new Date(Date.now() - monthlyRangeMs);
+		const count = await SETTINGS.countDocuments({ last_active: { $gt: cutoff } });
+		this.set(count);
+	}
+});
+
+export const activeYearlyGauge = new Gauge({
+	name: 'juxtaposition_active_users_yearly',
+	help: 'Users active in the last 365 days',
+	async collect(): Promise<void> {
+		const yearlyRangeMs = 365 * 24 * 60 * 60 * 1000;
+		const cutoff = new Date(Date.now() - yearlyRangeMs);
+		const count = await SETTINGS.countDocuments({ last_active: { $gt: cutoff } });
```

Comment:
How often will Grafana poll this? This query seems kinda expensive and the yearly average should change very slowly..

Reply:
My 2 cents, I agree with Quarky. Using an aggregate doesn't substantially change the code/complexity and gives us a free optimization, so I don't see a reason not to. The aggregate is arguably less clear than seeing `countDocuments` when reading things at a glance, but I'd expect anyone reading our metrics code should have a grasp on Mongo/Mongoose anyway (plus a little comment just noting that it's basically a faster `countDocuments` also solves that)